robocorp-code/vscode-client/src/extension.ts

```diff
@@ -324,6 +326,102 @@ async function cloudLogoutAndRefresh() {
     refreshCloudTreeView();
 }
 
+async function ensureConvertBundle() {
+    const url = "https://downloads.robocorp.com/converter/latest/bundle.js";
+    let location = getExtensionRelativeFile("../../vscode-client/out/converterBundle.js", false);
+    if (!(await fileExists(location))) {
+        await window.withProgress(
+            {
+                location: vscode.ProgressLocation.Notification,
+                title: "Downloading converter bundle",
+                cancellable: false,
+            },
+            async (
+                progress: vscode.Progress<{ message?: string; increment?: number }>,
+                token: vscode.CancellationToken
+            ): Promise<string | undefined> => await download(url, progress, token, location)
+        );
+    }
+    return location;
+}
+
+async function convertProject() {
+    const DEFAULT_ERROR_MSG = `
+            Could not convert project.
+            Please check the output logs for more details.
+            `;
+
+    const DEFAULT_ERROR_STATUS = "Error while converting project.";
+
+    const converterLocation = await ensureConvertBundle();
+    try {
+        const converterBundle = require(converterLocation);
+        window.setStatusBarMessage("Project conversion has started...");
+
+        // let the user decide what type of project will be converted
+        const vendorMap = {
+            "Blue Prism": "blueprism",
+            "UiPath": "uipath",
+            "Automation Anywhere 360": "a360",
+        };
+        const items = Object.keys(vendorMap);
+        let selectedItem = await window.showQuickPick(items, {
+            "placeHolder": "Please select vendor of current project",
+            "canPickMany": false,
+            "ignoreFocusOut": false,
+        });
+        window.setStatusBarMessage(`Selected project type: ${selectedItem}`);
+        if (!selectedItem) {
+            window.setStatusBarMessage("Project conversion was canceled.");
+            return;
+        }
+
+        // let the user decide where the conversion result will be saved
+        const destination = await window.showOpenDialog({
+            title: "Select conversion destination",
+            openLabel: "Select Output Folder",
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+        });
+        if (!destination || destination.length == 0) {
+            window.setStatusBarMessage("Project conversion was canceled.");
+            return;
+        }
+
+        window.setStatusBarMessage("Project conversion is starting...");
+        const activeTextEditor = window.activeTextEditor;
+        const conversionResult = await converterBundle.convert(
```

Comment:
It'd be nice to have some interface for the `conversionResult` to know what is actually there...